fedn/network/storage/statestore/stores/client_store.py

```
@@ -190,7 +190,7 @@ def update(self, id: str, item: Client) -> Tuple[bool, Any]:
             existing_item = session.scalars(stmt).first()
 
             if existing_item is None:
-                raise EntityNotFound(f"Entity with (id | client_id) {id} not found")
+                return False, "Item not found"
```

The update methods are possibly the one place we could raise the exception... I don't know what is best, I think both solutions work. We could perhaps take 2 min and talk it through...